### PR TITLE
ISPN-14213: Add a statement why we provide no performance numbers

### PR DESCRIPTION
**Pull request by dfitzmau**
JIRA: https://issues.redhat.com/browse/ISPN-14213

### CODE REVIEW

**Comment by oraNod** [comment]
@dfitzmau Does this need backport to 13.x?

File: documentation/src/main/asciidoc/stories/assembly_deployment_planning.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/stories/assembly_deployment_planning.adoc:36]:
```suggestion
Given the number of different combinations and unknown external factores, it is not possible to provide a performance calculation that meets all {brandname} use cases.
The purpose of this document, rather, is to provide details about use cases and architectures to help you determine requirements for your {brandname} deployment.
```

File: documentation/src/main/asciidoc/stories/assembly_deployment_planning.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/stories/assembly_deployment_planning.adoc:37]: I'd remove the downstream conditional here. That statement applies to community too.

**Review by oraNod — CHANGES_REQUESTED** [review]

**Comment by dfitzmau** [comment]
> @dfitzmau Does this need backport to 13.x?

Hi @oraNod .

I'll need to wait for Wolf to return from PTO to confirm this. 

@wfink

File: documentation/src/main/asciidoc/stories/assembly_deployment_planning.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/stories/assembly_deployment_planning.adoc]: Thanks for the honest feedback. I updated the content to state:

> The previous listed factors can lead to an uncountable combination of performance limitations that makes determining performance metrics for calculating optimum system configuration and workload thresholds difficult. 

WDYT?

**Review by wfink — APPROVED** [review]
looks good to me

**Comment by dfitzmau** [comment]
> > @dfitzmau Does this need backport to 13.x?
> 
> Hi @oraNod .
> 
> I'll need to wait for Wolf to return from PTO to confirm this.
> 
> @wfink

Hi @oraNod . Yes, I need to apply the same update to 8.3. I created another Jira to track similar 8.3 updates. You OK to approve this PR?

Any idea of the easiest way to backport changes to 8.3?

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:1]: I don't think this is necessary if the file name has the `con_` prefix.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:17]: Sentence is way too long.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:19]: I would move this sentence to the top. Maybe the opening sentence.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:21]: "from the {brandname} command line interface (CLI)"

Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:21]: Also please take a look at the one sentence per line recommendation. This is something the Infinispan doc set adheres to. 

https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line

**Review by oraNod — CHANGES_REQUESTED** [review]

**Comment by oraNod** [comment]
@dfitzmau A few more nits. When it comes to the backport, create a new branch that tracks to 13.0.x. Git cherry-pick the commit(s) on to that branch.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:21]: Hi @oraNod . This recommendation is based on a document in draft, and the style does not seem to be evident on other projects I've worked on. I do find sticking to this recommendation difficult, because I don't know the "why" of it? The principle has not been explained to me very well. Is it recommended because it makes scanning an _.adoc_ file easier?

Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:21]: The fact that doc is in some kind of draft state doesn't really matter. The advantages that doc lists hold true and explain the why.

In addition to all that, one sentence per line helps with PR reviews. But, generally, it also makes a lot of sense because of [how asciidoc creates paragraphs](https://docs.asciidoctor.org/asciidoc/latest/blocks/paragraphs/).

You can find this in the Infinispan contributor's guide here: https://infinispan.org/docs/stable/titles/contributing/contributing.html#documentation_guidelines_general

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: Moved to start of module.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: Split into two sentences.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: Removed.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:19]: Thanks for sharing, @oraNod . I really need to know "why", as I've been asked to implement things and being told they don't break anything, so why should they hold up the progress of a PR. 

I generally follow internal guidelines, so I can stay up-to-date with the "one voice". Who updates the Infinispan Contributor's guide? For example, "contrctions" have been relaxed. I generally don't use them, but they are no longer slapped with the non-recommended guidelines.

**Comment by dfitzmau** [comment]
> @dfitzmau A few more nits. When it comes to the backport, create a new branch that tracks to 13.0.x. Git cherry-pick the commit(s) on to that branch.

Thanks, @oraNod . I updated the .adoc file based on all your feedback.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: @dfitzmau little things like this can have larger, cumulative effects down the road. Part of why I brought it up is that I see using ventilated prose (one sentence per line) as a good writing habit. It's not just a style guide thing.

I picked it up from one of the core devs on this project and try to evangelize it a little. I'd suggest going back and reading Dan Allen's take on the advantages. He pretty much says it all.

I'm not sure what you mean by "one voice"?

"Who updates the Infinispan Contributor's guide?" - The Infinispan team update the Contributor's Guide.
"For example, "contrctions" have been relaxed. I generally don't use them, but they are no longer slapped with the non-recommended guidelines." - Again, not sure what you mean by this.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:20]:
```suggestion
You can run basic performance tests with the {brandname} CLI that collects limited performance metrics. 
```

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:21]:
```suggestion
You can customize the performance test so that the test outputs results that meet your needs. 
```

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:22]:
```suggestion
Test results provide baseline metrics that can help you determine settings and resources for your {brandname} caching requirements. 
```

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:25]: Honestly I think we can drop these two lines.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: I think the topic of "preference" is highly debatable. Chat channels are devoted to what someone thinks is right or wrong. I prefer consistency, so that one can remain agile among products.

"simplifying" processes was also mentioned at a very high-level. Having too many sources to reference will eventually lead to confusion, as they start to contradict one another. 

The contributor guide you linked mentions "Avoid contractions. Use "do not" instead of "don’t".",

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:21]: :heavy_check_mark:

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: :heavy_check_mark:

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:25]: I'll leave these sentences in because the SME requested them. These sentences might be obvious, but we really want to enforce that the user takes responsibility for determining their performance metrics. We cannot provide a "one-shot" performance metric that spans all customer user cases. Providing users with as much baseline help might help their reach their desired metric determination goals.

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc]: :heavy_check_mark:

**Review by oraNod — APPROVED** [review]
Thanks @dfitzmau

**Comment by dfitzmau** [comment]
> Thanks @dfitzmau

Thanks, @oraNod . I'm OK to merge? Based on your time on Infinispan, do I need three approvals before I can merge a PR?

**Comment by oraNod** [comment]
@dfitzmau Just waiting on CI

**Comment by dfitzmau** [comment]
> @dfitzmau Just waiting on CI

OK. Do you get notified when the CI process completes? How does one check or confirm?

**Comment by oraNod** [comment]
Merging as CI fails only on JUnit tests that don't really apply to docs. @ryanemerson Do you think it's worth excluding docs only commits from CI runs?

File: documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc
Comment (dvagnero) [inline documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc:5]: Hi! This is the community title but in the product it's different. This will require a follow up PR. Either you can reword it so you don't mention the title name or you can create an attribute for the title that will have different value in the community and in the product. I can help

**Pull request merged**
